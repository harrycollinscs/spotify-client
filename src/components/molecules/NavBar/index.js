import React, { useState } from "react";
import styled from "styled-components";
import { Link, NavLink } from "react-router-dom";
import { faCompactDisc } from "@fortawesome/free-solid-svg-icons";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import CircleImage from "../../atoms/CircleImage";
import store from "../../../store";

const StyledNavBar = styled.nav`
  width: 100%;
  padding: 10px;
  position: sticky;
  top: 0;
  background-color: #212121;
  display: flex;
  flex-direction: row;
  align-items: center;
  text-align: center;
  justify-content: space-between;
  z-index: 1000;
`;

const BrandAndLinksContainer = styled.div`
  display: flex;
  flex-direction: row;
`;

const BrandLogoContainer = styled.span`
  flex-direction: row;
  display: flex;
  padding: 10px 30px;
  align-items: center;
`;

const StyledBrand = styled.h2`
  color: white;
  align-self: center;
  font-weight: bold;
  margin: 5px
`;

const StyledNavLinks = styled.div`
  align-self: center;
  justify-content: space-between;
  margin: 0px 20px;
`;

const StyledListItem = styled.span`
  text-align: center;
  margin-right: 40px;
`;

const NavLinkActiveStyle = { 
  textDecoration: "none",
  color: "white" 
};

const NavLinkInactiveStyle = {
  textDecoration: 'none',
  color: '#b3b3b3',
}

const IconStyle = {
  color: '#1db954',
  margin: '5px',
  height: '25px',
  width: '25px',
};

const NavigationBar = ({ links, isLoggedIn }) => {

  return (
    <StyledNavBar>
      <BrandAndLinksContainer style={{display:'flex', flexDirection: 'row'}}>
      <Link to="/" style={{ textDecoration: 'none' }} >
        <BrandLogoContainer>
          <StyledBrand>Statify</StyledBrand>
          <FontAwesomeIcon icon={faCompactDisc} style={IconStyle}/>
        </BrandLogoContainer>
      </Link>
      <StyledNavLinks>
        {links.map((link) => (
          <StyledListItem key={link.title}>
            <NavLink
              to={`${link.route}`}
              style={NavLinkInactiveStyle}
              activeStyle={NavLinkActiveStyle}
            >
              {link.title}
            </NavLink>
          </StyledListItem>
        ))}
      </StyledNavLinks>
      </BrandAndLinksContainer>
      { store.getState().user && <CircleImage style={{float:'right'}} url={store.getState().user.images[0].url}/>}
    </StyledNavBar>
  );
};

export default NavigationBar;
