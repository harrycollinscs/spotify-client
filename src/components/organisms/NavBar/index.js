import React from "react";
import styled from "styled-components";
import { Link, NavLink } from "react-router-dom";
import { faSpotify } from "@fortawesome/free-brands-svg-icons";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import CircleImage from "../../atoms/CircleImage";
import store from "../../../store";

const StyledNavBar = styled.nav`
  width: 100vw;
  height: 8vh;
  padding: 20px 0px;
  background-color: #000;
  align-items: center;
  text-align: center;
  z-index: 1000;
  display: flex;
  position: fixed;
  top: 0;
  flex-direction: row;
  justify-content: space-between;
`;

const BrandLogoContainer = styled.span`
  flex-direction: row;
  display: flex;
  align-items: center;
  text-align: left;
  align-content: center;
  padding: 10px;
`;

const StyledBrand = styled.h3`
  color: white;
  align-self: center;
  font-weight: bold;
  margin: 5px;
`;

const StyledNavLinks = styled.div`
  margin: 30px 0px;
  text-align: flex-start;
  padding: 0px;
  display: flex;
  flex-direction: row;
`;

const StyledListItem = styled.div`
  text-align: left;
  margin: 5px 0px;
  padding: 0px 20px;
`;

const IconContainer = styled.div`
  display: flex;
  justify-content: space-evenly;
`;

const NavLinkActiveStyle = {
  textDecoration: "none",
  fontWeight: "bold",
  color: "white",
};

const NavLinkInactiveStyle = {
  textDecoration: "none",
  color: "grey",
};

const IconStyle = {
  color: "#fff",
  margin: "5px",
  height: "25px",
  width: "25px",
};

const NavigationBar = ({ links }) => {
  const { user } = store.getState();

  return (
    <StyledNavBar>
      <Link to="/songs" style={{ textDecoration: "none" }}>
        <BrandLogoContainer>
          <StyledBrand>Spotify</StyledBrand>
          <FontAwesomeIcon icon={faSpotify} style={IconStyle} />
        </BrandLogoContainer>
      </Link>

      <StyledNavLinks>
        {links.map((link) => (
          <StyledListItem key={link.title}>
            <NavLink
              to={`${link.route}`}
              style={NavLinkInactiveStyle}
              activeStyle={NavLinkActiveStyle}
            >
              {link.title}
            </NavLink>
          </StyledListItem>
        ))}
      </StyledNavLinks>

      <IconContainer>
        {user && <CircleImage url={store.getState().user.images[0].url} />}
      </IconContainer>
    </StyledNavBar>
  );
};

export default NavigationBar;
