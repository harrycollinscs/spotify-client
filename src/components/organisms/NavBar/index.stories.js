import React from "react";
import NavBar from "./index";

export default {
  title: "Organisms/NavBar",
  component: NavBar,
};

const Links = [
  {
    title: "Songs",
    route: "/songs",
  },
  {
    title: "Playlists",
    route: "/playlists",
  },
];

const Template = (args) => <NavBar {...args} />;

export const Default = Template.bind({});
Default.args = {
  links: Links,
};
