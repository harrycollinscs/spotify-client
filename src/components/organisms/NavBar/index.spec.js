import React from "react";
import NavBar from "./index";
import { shallow } from "enzyme";

describe("Organisms/NavBar", () => {
  const props = {
    links: [
      {
        title: "Songs",
        route: "/songs",
      },
      {
        title: "Artists",
        route: "/artists",
      },
      {
        title: "Playlists",
        route: "/playlists",
      },
    ],
  };

  const component = <NavBar {...props} />;

  it("Should match snapshot", () => {
    const render = shallow(component);
    expect(render).toMatchSnapshot();
  });
});
