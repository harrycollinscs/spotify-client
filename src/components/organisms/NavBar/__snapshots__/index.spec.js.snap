// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Organisms/NavBar Should match snapshot 1`] = `
<styled.nav>
  <Link
    style={
      Object {
        "textDecoration": "none",
      }
    }
    to="/"
  >
    <styled.span>
      <styled.h2>
        Reactify
      </styled.h2>
      <FontAwesomeIcon
        border={false}
        className=""
        fixedWidth={false}
        flip={null}
        icon={
          Object {
            "icon": Array [
              496,
              512,
              Array [],
              "f51f",
              "M248 8C111 8 0 119 0 256s111 248 248 248 248-111 248-248S385 8 248 8zM88 256H56c0-105.9 86.1-192 192-192v32c-88.2 0-160 71.8-160 160zm160 96c-53 0-96-43-96-96s43-96 96-96 96 43 96 96-43 96-96 96zm0-128c-17.7 0-32 14.3-32 32s14.3 32 32 32 32-14.3 32-32-14.3-32-32-32z",
            ],
            "iconName": "compact-disc",
            "prefix": "fas",
          }
        }
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        style={
          Object {
            "color": "#fff",
            "height": "25px",
            "margin": "5px",
            "width": "25px",
          }
        }
        swapOpacity={false}
        symbol={false}
        title=""
        transform={null}
      />
    </styled.span>
  </Link>
  <styled.div>
    <styled.div
      key="Songs"
    >
      <NavLink
        activeStyle={
          Object {
            "color": "white",
            "fontWeight": "bold",
            "textDecoration": "none",
          }
        }
        style={
          Object {
            "color": "grey",
            "textDecoration": "none",
          }
        }
        to="/songs"
      >
        Songs
      </NavLink>
    </styled.div>
    <styled.div
      key="Artists"
    >
      <NavLink
        activeStyle={
          Object {
            "color": "white",
            "fontWeight": "bold",
            "textDecoration": "none",
          }
        }
        style={
          Object {
            "color": "grey",
            "textDecoration": "none",
          }
        }
        to="/artists"
      >
        Artists
      </NavLink>
    </styled.div>
    <styled.div
      key="Playlists"
    >
      <NavLink
        activeStyle={
          Object {
            "color": "white",
            "fontWeight": "bold",
            "textDecoration": "none",
          }
        }
        style={
          Object {
            "color": "grey",
            "textDecoration": "none",
          }
        }
        to="/playlists"
      >
        Playlists
      </NavLink>
    </styled.div>
  </styled.div>
  <styled.div />
</styled.nav>
`;
